feat(layers): add support for publishing v2 layer

rubenfonseca opened the pull request:
**Issue number:** #1379

## Summary

This PR adds support for publishing the v2 layer of Lambda Powertools for Python

### Changes

> Please provide a summary of what's being changed

- Created separate workflows for v2 (so we can keep v1 running)
- Moved all additional dependencies to a new "all" group
- Marked boto3 as dev dependency
- Changed Layer CDK code to produce and canary test 2 layers (regular + arm64)
- Configured github actions to be able to produce arm64 (via qemu emulation)

### User experience

> Please share what the user experience looks like before and after this change

After merging this we should be able to start testing the v2 layer on the beta branch.

In order for it to work, we need a new major release of the [CDK Layer Construct](https://github.com/awslabs/cdk-aws-lambda-powertools-layer/pull/4)

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [ ] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

rubenfonseca committed:
feat(layers): add initial support for publishing v2 beta layer

rubenfonseca committed:
chore(layer): rename workflow

rubenfonseca committed:
chore(deps): moved extra dependencies into an extra poetry group

rubenfonseca committed:
feat(layers): add regular and arm64 flavours

rubenfonseca committed:
chore(deps): move fastjsonschema to optinal dependency group

rubenfonseca committed:
feat(layers): use qmeu for arm64 builds

rubenfonseca committed:
fix(layers): pinned github actions

heitorlessa commented on pyproject.toml:
```diff
@@ -76,7 +76,7 @@ filelock = "^3.8.0"
 
 [tool.poetry.extras]
-pydantic = ["pydantic", "email-validator"]
+extras = ["pydantic", "email-validator", "aws-xray-sdk", "fastjsonschema"]
```
Can we rename it to `all` and also have a group per each feature e.g. `tracer`, `parser`, etc.?

rubenfonseca committed:
chore: create extra dependency groups per utility

heitorlessa commented on .github/workflows/publish_v2_layer.yml:
```diff
@@ -0,0 +1,91 @@
+  workflow_dispatch:
+    inputs:
+      latest_published_version:
+        description: "Latest PyPi published version to rebuild latest docs for, e.g. v1.22.0"
```
```suggestion
        description: "Latest PyPi published version to rebuild latest docs for, e.g. v2.0.0"
```

heitorlessa commented on .github/workflows/publish_v2_layer.yml:
```diff
@@ -0,0 +1,91 @@
+    inputs:
+      latest_published_version:
+        description: "Latest PyPi published version to rebuild latest docs for, e.g. v1.22.0"
+        default: "v1.22.0"
```
```suggestion
        default: "v2.0.0"
```

heitorlessa commented on .github/workflows/reusable_deploy_v2_layer_stack.yml:
```diff
@@ -0,0 +1,99 @@
+    strategy:
+      fail-fast: false
+      matrix:
+        region:
```
this will fail when we publish ARM as it's not available for all regions yet - Not entirely sure where this data is kept either.

heitorlessa replied:
Can be resolved as we discussed the use of Pricing API (separate PR on a scheduled job)

heitorlessa commented on .github/workflows/reusable_deploy_v2_layer_stack.yml:
```diff
@@ -0,0 +1,99 @@
+          pip install -r requirements.txt
+      - name: install cdk and deps
+        run: |
+          npm install -g aws-cdk@2.29.0
```
```suggestion
          npm install -g aws-cdk@2.44.0
```

syncing both workflows version

heitorlessa commented on .github/workflows/reusable_deploy_v2_layer_stack.yml:
```diff
@@ -0,0 +1,99 @@
+permissions:
+  id-token: write
+  contents: read
+
```
```yaml
env:
    CDK_VERSION: 2.44.0
```


For our future selves

heitorlessa commented on Makefile:
```diff
@@ -6,12 +6,12 @@ target:
 dev:
 	pip install --upgrade pip pre-commit poetry
-	poetry install --extras "pydantic"
+	poetry install --extras "extras"
```
```suggestion
	poetry install --extras "all"
```

heitorlessa commented on Makefile:
```diff
@@ -6,12 +6,12 @@ target:
 dev-gitpod:
 	pip install --upgrade pip poetry
-	poetry install --extras "pydantic"
+	poetry install --extras "extras"
```
```suggestion
	poetry install --extras "all"
```

heitorlessa commented on layer/app.py:
```diff
@@ -21,13 +22,15 @@
     "LayerStack",
     powertools_version=POWERTOOLS_VERSION,
     ssm_paramter_layer_arn=SSM_PARAM_LAYER_ARN,
+    ssm_parameter_arm64_layer_arn=SSM_PARAM_ARM64_LAYER_ARN,
```
```suggestion
    ssm_parameter_layer_arm64_arn=SSM_PARAM_LAYER_ARM64_ARN,
```

heitorlessa commented on layer/app.py:
```diff
@@ -21,13 +22,15 @@
     "CanaryStack",
     powertools_version=POWERTOOLS_VERSION,
     ssm_paramter_layer_arn=SSM_PARAM_LAYER_ARN,
+    ssm_parameter_arm64_layer_arn=SSM_PARAM_ARM64_LAYER_ARN,
```
```suggestion
    ssm_parameter_layer_arm64_arn=SSM_PARAM_LAYER_ARM64_ARN,
```

heitorlessa commented on layer/layer/canary/app.py:
```diff
@@ -66,7 +67,7 @@ def verify_powertools_version() -> None:
     if powertools_version != current_version:
         raise ValueError(
-            f'Expected powertoosl version is "{powertools_version}", but layer contains version "{current_version}"'
+            f'Expected powertools version is "{powertools_version}", but layer contains version "{current_version}"'
```
```suggestion
            f'Expected Powertools version is "{powertools_version}", but layer contains version "{current_version}"'
```

heitorlessa commented on layer/layer/canary_stack.py:
```diff
@@ -16,29 +20,70 @@ def __init__(
         construct_id: str,
         powertools_version: str,
         ssm_paramter_layer_arn: str,
+        ssm_parameter_arm64_layer_arn: str,
```
```suggestion
        ssm_parameter_layer_arm64_arn: str,
```

heitorlessa commented on layer/layer/canary_stack.py:
```diff
@@ -16,29 +20,70 @@ def __init__(
+        layer_arm64_arn = StringParameter.from_string_parameter_attributes(
+            self,
+            "LayerArm64VersionArnParam",
+            parameter_name=ssm_parameter_arm64_layer_arn,
```
```suggestion
            parameter_name=ssm_parameter_layer_arm64_arn,
```

heitorlessa commented on layer/layer/canary_stack.py:
```diff
@@ -49,7 +94,8 @@ def __init__(
             memory_size=512,
             timeout=Duration.seconds(10),
-            runtime=Runtime.PYTHON_3_9,
+            runtime=Runtime.PYTHON_3_9(),
```
```suggestion
            runtime=Runtime.PYTHON_3_9,
```

ref: https://github.com/awslabs/aws-lambda-powertools-python/blob/c31f5138b5fa2bac589c7e3b58d236abe536bb6c/tests/e2e/utils/infrastructure.py#L109

heitorlessa commented on layer/layer/layer_stack.py:
```diff
@@ -1,18 +1,38 @@
+        construct_id: str,
+        powertools_version: str,
+        ssm_paramter_layer_arn: str,
+        ssm_parameter_arm64_layer_arn: str,
```
```suggestion
        ssm_parameter_layer_arm64_arn: str,
```

heitorlessa commented on layer/layer/layer_stack.py:
```diff
@@ -23,9 +43,32 @@ def __init__(
+        StringParameter(
+            self,
+            "Arm64VersionArn",
+            parameter_name=ssm_parameter_arm64_layer_arn,
```
```suggestion
            parameter_name=ssm_parameter_layer_arm64_arn,
```

heitorlessa commented on layer/layer/layer_stack.py:
```diff
@@ -23,9 +43,32 @@ def __init__(
+        )
 
         CfnOutput(self, "LatestLayerArn", value=layer.layer_version_arn)
+        CfnOutput(self, "LatestArm64LayerArn", value=layer_arm64.layer_version_arn)
```
```suggestion
        CfnOutput(self, "LatestLayerArm64Arn", value=layer_arm64.layer_version_arn)
```

heitorlessa commented on layer/app.py:
```diff
@@ -8,7 +8,8 @@
 POWERTOOLS_VERSION: str = app.node.try_get_context("version")
-SSM_PARAM_LAYER_ARN: str = "/layers/powertools-layer-arn"
+SSM_PARAM_LAYER_ARN: str = "/layers/powertools-layer-v2-arn"
+SSM_PARAM_ARM64_LAYER_ARN: str = "/layers/powertools-layer-v2-arm64-arn"
```
```suggestion
SSM_PARAM_LAYER_ARM64_ARN: str = "/layers/powertools-layer-v2-arm64-arn"
```

heitorlessa commented on pyproject.toml:
```diff
@@ -76,7 +76,10 @@ filelock = "^3.8.0"
+validation = ["fastjsonschema"]
+tracer = ["aws-xray-sdk"]
+all = ["pydantic", "email-validator", "aws-xray-sdk", "fastjsonschema"]
 
```
we need to take into account `jmespath` too. 

`jmespath = ["jmespath"]` 

Every utility providing an `envelope` uses `jmespath` under the hood... which came with `boto3` or `aws-xray-sdk`. For example:  `parser`, `idempotency`, etc.

`idempotency = ["jmespath"]`

heitorlessa commented on pyproject.toml:
```diff
@@ -76,7 +76,10 @@ filelock = "^3.8.0"
 [tool.poetry.extras]
-pydantic = ["pydantic", "email-validator"]
+parser = ["pydantic", "email-validator"]
+validation = ["fastjsonschema"]
```
```suggestion
validation = ["fastjsonschema", "jmespath"]
```

heitorlessa commented on pyproject.toml:
```diff
@@ -76,7 +76,10 @@ filelock = "^3.8.0"
 
 [tool.poetry.extras]
-pydantic = ["pydantic", "email-validator"]
+parser = ["pydantic", "email-validator"]
```
```suggestion
parser = ["pydantic", "email-validator", "jmespath"]
```

heitorlessa reviewed (changes requested):
as per peer review session

rubenfonseca committed:
chore: apply suggestions from code review

Co-authored-by: Heitor Lessa <lessa@amazon.com>

rubenfonseca committed:
chore: use latest cdk layer construct

rubenfonseca committed:
chore: add jmespath and idempotency deps groups

rubenfonseca committed:
chore: lock deps

rubenfonseca committed:
chore: use new cdk construct name

rubenfonseca committed:
chore: update call to CDK architecture

rubenfonseca committed:
chore: fix canary deployment

rubenfonseca committed:
feat: use canary to test for presence of optional dependencies

rubenfonseca committed:
chore: bump cdk to 2.44 so we can use direct mode

rubenfonseca committed:
chore: typo

rubenfonseca committed:
feat: use CDK direct mode on e2e and remove unecessary deps

rubenfonseca committed:
chore: remove badly escaped strings

heitorlessa commented on .github/workflows/publish_v2_layer.yml:
```diff
@@ -0,0 +1,91 @@
+          echo RELEASE_TAG_VERSION="${RELEASE_TAG_VERSION:1}" >> "$GITHUB_ENV"
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@8b122486cedac8393e77aa9734c3528886e4a1a8 # v2.0.0
+      - name: Set up Docker Buildx
```
```suggestion
      - name: Set up QEMU
        uses: docker/setup-qemu-action@8b122486cedac8393e77aa9734c3528886e4a1a8 # v2.0.0
        # NOTE: we need QEMU to build Layer against a different architecture (e.g., ARM)
      - name: Set up Docker Buildx
```

heitorlessa commented on .github/workflows/publish_v2_layer.yml:
```diff
@@ -0,0 +1,91 @@
+      artefact-name: "cdk-layer-artefact"
+      environment: "layer-beta"
+
+  # deploy-prod:
```
Note to ourselves: we need to create a launch plan in the v2 RFC and remember to uncomment deploy-prod

rubenfonseca replied:
Added to the v2 RFC

heitorlessa commented on layer/layer/canary/app.py:
```diff
@@ -17,6 +22,26 @@
 event_bus_arn = os.getenv("VERSION_TRACKING_EVENT_BUS_ARN")
 
 
+# Model to check parser imports correctly, tests for pydantic and email-validator
+class OrderItem(BaseModel):
+    order_id: int
+    quantity: int
+    description: str
+    email: EmailStr
+
+
+# Tests for jmespath presence
+@event_parser(model=OrderItem, envelope=envelopes.EventBridgeEnvelope)
+def envelope_handler(event: OrderItem, context: LambdaContext):
+    assert event.order_id != 1
+
+
+# Tests for fastjsonschema presence
+@validator(inbound_schema={}, envelope="detail")
+def validator_handler(event, context: LambdaContext):
+    pass
+
+
```
Amazing work and comments here!

heitorlessa commented on layer/layer/canary/app.py:
```diff
@@ -66,7 +91,7 @@ def verify_powertools_version() -> None:
     if powertools_version != current_version:
         raise ValueError(
-            f'Expected powertoosl version is "{powertools_version}", but layer contains version "{current_version}"'
+            f'Expected Powertools version is "{powertools_version}", but layer contains version "{current_version}"'
```
😈

heitorlessa commented on layer/layer/canary/app.py:
```diff
@@ -80,6 +105,8 @@ def send_notification():
         )
         return
+
+    architecture = platform.uname()[4]
```
```suggestion
    architecture = platform.uname()[4]  # arm64, x86_64
```

heitorlessa commented on layer/layer/canary/app.py:
```diff
@@ -90,6 +117,7 @@ def send_notification():
                 "version": powertools_version,
                 "region": os.environ["AWS_REGION"],
                 "layerArn": layer_arn,
+                "architecture": architecture,
```
Thank you for thinking ahead and storing this piece of info there.

heitorlessa commented on tests/e2e/utils/lambda_layer/powertools_layer.py:
```diff
@@ -16,9 +16,16 @@ class LocalLambdaPowertoolsLayer(BaseLocalLambdaLayer):
         self.build_args = "--platform manylinux1_x86_64 --only-binary=:all: --upgrade"
         self.build_command = f"python -m pip install {self.package} {self.build_args} --target {self.target_dir}"
+        self.cleanup_command = (
+            f"rm -rf {self.target_dir}/boto* {self.target_dir}/s3transfer* && "
+            f"rm -rf {self.target_dir}/*dateutil* {self.target_dir}/urllib3* {self.target_dir}/six* && "
```
```suggestion
            f"rm -rf {self.target_dir}/boto* {self.target_dir}/s3transfer* && "
            f"rm -rf {self.target_dir}/*dateutil* {self.target_dir}/urllib3* {self.target_dir}/six* && "
            f"rm -rf {self.target_dir}/jmespath* && "
```

We forgot `jmespath` as it comes with [`botocore`](https://github.com/boto/botocore/blob/develop/setup.py#L26). This means we don't need that as an optional either (I'll comment in `pyproject.toml` directly)

rubenfonseca replied:
Note to self: apply this to the cdk construct too

heitorlessa commented on tests/e2e/utils/lambda_layer/powertools_layer.py:
```diff
@@ -31,6 +38,11 @@ def build(self) -> str:
         return str(self.output_dir)
 
+    def after_build(self):
+        subprocess.run(self.cleanup_command, shell=True)
```
future note (not this PR): we should look to make it work without `shell=True`. Typically all you need is a list of commands, but I had issues with Layer build last time.

heitorlessa commented on tests/e2e/utils/lambda_layer/powertools_layer.py:
```diff
@@ -31,6 +38,11 @@ def build(self) -> str:
+    def after_build(self):
+        subprocess.run(self.cleanup_command, shell=True)
+
+        super(LocalLambdaPowertoolsLayer, self).after_build()
```
I don't think we need that? Please correct me if I'm wrong.

We're implementing `BaseLocalLambdaLayer` ABC, and specifically `after_build`.

If a future class subclass `LocalLambdaPowertoolsLayer` and wants to extend `after_build`, then this would be valid.

rubenfonseca replied:
we don't, I was just trying to be a good citizen :) Deleting it

heitorlessa commented on tests/e2e/utils/lambda_layer/powertools_layer.py:
```diff
@@ -16,9 +16,16 @@ class LocalLambdaPowertoolsLayer(BaseLocalLambdaLayer):
         super().__init__(output_dir)
-        self.package = f"{SOURCE_CODE_ROOT_PATH}[pydantic]"
+        self.package = f"{SOURCE_CODE_ROOT_PATH}[all]"
         self.build_args = "--platform manylinux1_x86_64 --only-binary=:all: --upgrade"
```
We're gonna need to extract the platform argument into a separate variable, so we can determine the correct value:

* **X86**: `manylinux1_x86_64`
* **ARM** `manylinux2014_aarch64`

It took me a while to figure out the oldest OS wheel distro supporting ARM that would work for X-Ray SDK dependency (`wrapt`).

My repro:

**ARM**

```bash
pip install .[all] --platform manylinux2014_aarch64 --only-binary=:all: -t /tmp/ptARM
ls -l /tmp/ptARM/wrapt/
```

**X86**

```bash
pip install .[all] --platform manylinux1_x86_64 --only-binary=:all: -t /tmp/ptX86
ls -l /tmp/ptX86/wrapt/
```

heitorlessa commented on pyproject.toml:
```diff
@@ -77,7 +77,12 @@ checksumdir = "^1.2.0"
+parser = ["pydantic", "email-validator", "jmespath"]
+validation = ["fastjsonschema", "jmespath"]
+tracer = ["aws-xray-sdk"]
+idempotency = ["jmespath"]
+jmespath = ["jmespath"]
```
```suggestion
```

I was wrong. We're not gonna need that anymore because it comes with `botocore`. If boto ever removes we'll catch it in our E2E tests now.

This wasn't working either because we didn't specify the dependency `jmespath` like we did for `pydantic` and others.

heitorlessa commented on pyproject.toml:
```diff
@@ -77,7 +77,12 @@ checksumdir = "^1.2.0"
 
 [tool.poetry.extras]
-pydantic = ["pydantic", "email-validator"]
+parser = ["pydantic", "email-validator", "jmespath"]
+validation = ["fastjsonschema", "jmespath"]
```
```suggestion
parser = ["pydantic", "email-validator"]
validation = ["fastjsonschema"]
```

I was wrong. We're not gonna need that anymore because it comes with botocore. If boto ever removes we'll catch it in our E2E tests now.

This wasn't working either because we didn't specify the dependency `jmespath` like we did for `pydantic` and others.

heitorlessa commented on pyproject.toml:
```diff
@@ -20,18 +20,18 @@ license = "MIT-0"
+aws-xray-sdk = { version = "^2.8.0", optional = true }
+fastjsonschema = { version = "^2.14.5", optional = true }
+pydantic = { version = "^1.8.2", optional = true }
+email-validator = { version = "*", optional = true }
```
```suggestion
email-validator = { version = "^1.3.0", optional = true }
```

Sins from the past. This will make our install slightly faster, since `*` will instruct pip to download all available options in PyPi before handling the dependency resolution.  That is, it'll download everything listed here before choosing one: https://pypi.org/project/email-validator/#history

heitorlessa commented on layer/layer/canary_stack.py:
```diff
@@ -16,29 +20,74 @@ def __init__(
         execution_role.add_to_policy(
-            PolicyStatement(effect=Effect.ALLOW, actions=["lambda:GetFunction"], resources=["*"])
+            PolicyStatement(
+                effect=Effect.ALLOW, actions=["lambda:GetFunction"], resources=["*"]
```
@am29d please remind me again why do we need this permission in the Lambda itself? leftover?

Looking at the new canary I don't see an API call that would require this permission.

heitorlessa reviewed (changes requested):
Some quick suggestions, praises (my god, amazing work), and two issues spotted: **1/** Pip is bringing a wheel for X86 despite building for ARM (works but slower), **2/** Sin from the past on how we mistakenly accepted a PR with `email-validator` where we're downloading all available versions before choosing one.

heitorlessa commented:
Disregard my comment on JMESPath - we need to cater for PyPi customers too!

We just need to define the JMESPath as an optional dependency for extra to work

rubenfonseca committed:
chore: apply suggestions from code review

Co-authored-by: Heitor Lessa <lessa@amazon.com>

rubenfonseca committed:
chore: apply suggestions from code review

Co-authored-by: Heitor Lessa <lessa@amazon.com>

rubenfonseca committed:
chore: lock dependencies

rubenfonseca committed:
chore: remove unecessary code

rubenfonseca committed:
chore: apply suggestions from code review

Co-authored-by: Heitor Lessa <lessa@amazon.com>

rubenfonseca committed:
feat: allow e2e tests to run on arm64

rubenfonseca committed:
chore: add jmespath as an optional dependency

rubenfonseca committed:
chore: bumped layer construct version

rubenfonseca commented:
@heitorlessa ready to go again, incorporated your feedback, and added an code path to run our e2e tests using arm64 (no CLI flag was created yet)

heitorlessa commented on pyproject.toml:
```diff
@@ -23,11 +23,10 @@ python = "^3.7.4"
 pydantic = { version = "^1.8.2", optional = true }
-email-validator = { version = "*", optional = true }
+email-validator = { version = "^1.3.0", optional = true }
+jmespath = { version = "^1.0.1", optional = true }
```
```suggestion
boto3 = { version = "^1.20.32", optional = true }
```

heitorlessa commented on pyproject.toml:
```diff
@@ -77,11 +76,9 @@ checksumdir = "^1.2.0"
-idempotency = ["jmespath"]
-jmespath = ["jmespath"]
 all = ["pydantic", "email-validator", "aws-xray-sdk", "fastjsonschema"]
 
```
```suggestion
# allow customers to run code locally without emulators (SAM CLI, etc.)
aws-sdk = ["boto3"]
```

heitorlessa commented on tests/e2e/utils/infrastructure.py:
```diff
@@ -53,7 +53,9 @@ def __init__(self) -> None:
 
-    def create_lambda_functions(self, function_props: Optional[Dict] = None) -> Dict[str, Function]:
+    def create_lambda_functions(
+        self, function_props: Optional[Dict] = None, architecture: Optional[Architecture] = Architecture.X86_64
```
```suggestion
        self, function_props: Optional[Dict] = None, architecture: Architecture = Architecture.X86_64
```

heitorlessa commented on tests/e2e/utils/infrastructure.py:
```diff
@@ -65,6 +67,9 @@ def create_lambda_functions(self, function_props: Optional[Dict] = None) -> Dict
         function_props: Optional[Dict]
             Dictionary representing CDK Lambda FunctionProps to override defaults
 
+        architecture: Optional[Architecture]
```
```suggestion
        architecture: Architecture
```

heitorlessa commented on tests/e2e/utils/lambda_layer/powertools_layer.py:
```diff
@@ -58,3 +60,18 @@ def _has_source_changed(self) -> bool:
 
         return False
+
+    def _platform_name(self, architecture: Architecture) -> str:
```
```suggestion
    def _resolve_platform(self, architecture: Architecture) -> str:
```

heitorlessa commented on tests/e2e/utils/lambda_layer/powertools_layer.py:
```diff
@@ -14,14 +15,17 @@
         self.package = f"{SOURCE_CODE_ROOT_PATH}[all]"
-        self.build_args = "--platform manylinux1_x86_64 --only-binary=:all: --upgrade"
+
+        platform_name = self._platform_name(architecture)
```
```suggestion
        platform_name = self._resolve_platform(architecture)
```

heitorlessa replied:
pr: more explicit name

heitorlessa reviewed (changes requested):
last changes and it's good to go! (pinky promise)

heitorlessa committed:
chore(dep): add boto as optional dep

This will help customers prototyping locally without SAM CLI

heitorlessa committed:
chore(dep): add aws-sdk extras to ease local dev

This is due to jmespath and boto in general not being available locally, if a customer doesn't depend on boto3 as a dep (uses runtime)

heitorlessa committed:
chore(typing): remove unnecessary optional type

heitorlessa committed:
chore(typing): remove unnecessary optional type

heitorlessa committed:
refactor: use more explicit name for intent

heitorlessa committed:
chore: use more explicit name

rubenfonseca committed:
chore: lock dependencies

rubenfonseca committed:
chore: update cdk construct dependency
